gramps/plugins/lib/libmetadata.py

```diff
@@ -97,159 +112,284 @@ def format_gps(raw_dms, nsew):
 
-    def display_exif_tags(self, full_path):
+    def display_exif_tags(self, image_path):
         """
         Display the exif tags.
         """
         self.sections = {}
         self.model.clear()
-
-        if not os.path.exists(full_path):
+        self.selection_widget = SelectionWidget()
+        if not os.path.exists(image_path):
             return False
-
+        
         retval = False
-        with open(full_path, 'rb') as fd:
+        with open(image_path, 'rb') as fd:
             try:
                 buf = fd.read()
                 metadata = GExiv2.Metadata()
                 metadata.open_buf(buf)
-
+                mypixbuf = GdkPixbuf.Pixbuf.new_from_file(image_path)
                 get_human = metadata.get_tag_interpreted_string
+    
+                for section, key, key2, func, func2 in TAGS:
+                    if not key in metadata.get_exif_tags() + metadata.get_xmp_tags() + metadata.get_iptc_tags():
+                        continue
+
+                    if func is not None and func2 is not None:
+                        continue
 
-                for section, key, key2, func in TAGS:
-                    if not key in metadata.get_exif_tags():
+                    if func2 is not None:
+                        func2(self, metadata[key], metadata, image_path, mypixbuf)
                         continue
 
                     if func is not None:
                         if key2 is None:
                             human_value = func(metadata[key])
                         else:
-                            if key2 in metadata.get_exif_tags():
+                            if key2 in metadata.get_exif_tags() + metadata.get_xmp_tags() + metadata.get_iptc_tags():
                                 human_value = func(metadata[key], metadata[key2])
                             else:
                                 human_value = func(metadata[key], None)
                     else:
                         human_value = get_human(key)
-                        if key2 in metadata.get_exif_tags():
+                        if key2 in metadata.get_exif_tags() + metadata.get_xmp_tags() + metadata.get_iptc_tags():
                             human_value += ' ' + get_human(key2)
-
-                    label = metadata.get_tag_label(key)
-                    node = self.__add_section(section)
+                    
                     if human_value is None:
                         human_value = ''
-                    self.model.add((label, human_value), node=node)
-
+                    
+                    node = self.add_section(section)
+                    self.model.add([key, None, human_value], node=node)
+                    
                     self.model.tree.expand_all()
                     retval = self.model.count > 0
             except:
                 pass
 
         return retval
 
-    def __add_section(self, section):
+    def add_section(self, section):
```

Comment:
When it was private the gramplet wouldn't display any metadata.  I suspect because `get_named_regions `is not part of the class.  When I added `get_named_regions `into the class the code failed here:
https://github.com/brucejackson/gramps/blob/d24998743fad23b5a3d111896025f68327c34003/gramps/plugins/lib/libmetadata.py#L232

I could not figure out how to call `get_named_regions` that would cause an error.  Any pointers?

Reply:
I would convert `get_named_regions` into a method.  I would also get rid of the `func2` definitions.

Then in the loop evaluating the tags, add some code like:
```
if key == 'Xmp.mwg-rs.Regions':
    node = self.add_section(PEOPLE)
    for row in self.get_named_regions(self.pixbuf):
        self.model.add(row, node=node)
    break
```

I notice that you can also simplify the code for the rectangles.  You don't need to use the `SelectionWidget' method `proportional_to_real_rect`.  If you multiply the xmp value (a decimal between 0 an 1) by the width or height you will get the pixel value.

Reply:
I've moved the get_named_regions to __get_named_regions and did the same to the other functions.  I've hopefully improved things.

Reply:
Thanks.  It looks good now.